### PR TITLE
Bug 1273037 - Move from Personal to Google Auth

### PR DESCRIPTION
**Pull request by jezdez**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/telemetry-analysis-service/15)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by maurodoglio** [comment]
Please ping/mention me when this is ready for a review 😺

**Comment by jezdez** [comment]
@maurodoglio Okay, this seems good to review, the allauth templates that we're using right now don't match the style and may require some rewording here and there, I'll get to that tomorrow.

When you test this, you won't have to enter credentials separately, I've created a google client ID limited to localhost and it's being automatically added to the database (where allauth stores its creds) during the migrations.

File: atmo/users/adapters.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class AtmoSocialAccountAdapter(DefaultSocialAccountAdapter):
+
+    def validate_disconnect(self, account, accounts):
+        """
+        Validate whether or not the socialaccount account can be
+        safely disconnected.
+        """
+        if len(accounts) == 1:
+            raise forms.ValidationError(
+                'At least one account needs to be connected to your'
```
Comment (maurodoglio) [inline atmo/users/adapters.py]: s/ to your//

File: atmo/templates/atmo/dashboard.html [hunk 2]
```diff
@@ -27,7 +27,13 @@
         <li class="dropdown">
           <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">{{ request.user.email }} <span class="caret"></span></a>
           <ul class="dropdown-menu">
-            <li>{% browserid_logout text='Log Out' next='/login' %}</li>
+            {% if request.user.is_authenticated() %}
+            <li><a href="{{ url('account_email') }}">Emails</a></li>
```
Comment (maurodoglio) [inline atmo/templates/atmo/dashboard.html]: Actually nevermind, I noticed you fixed it in one of the commits below

File: atmo/templates/account/login.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    <h2>What is this?</h2>
+    <p>The Telemetry Data Analysis service gives direct access to Mozilla Telemetry data:</p>
+    <ul>
+      <li>Perform interactive analyses with <a href="http://spark.apache.org/">Apache Spark</a> and <a href="http://jupyter.org/">Jupyter</a>.</li>
+      <li>Run custom map-reduce analyses with the <a href="https://github.com/mozilla/telemetry-server/blob/master/docs/MapReduce.md">Telemetry MapReduce framework</a>.</li>
+      <li>Schedule these analyses so they run on a regular basis, publishing output to <a href="https://aws.amazon.com/s3/">Amason S3</a>.</li>
+    </ul>
+  </div>
+  <div class="col-md-6">
+    <h2>Links & resources</h2>
```
Comment (peterbe) [inline atmo/templates/account/login.html]: doesn't this need to be `&amp;`

File: atmo/users/migrations/0002_google_auth.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+def add_socialapp(apps, schema_editor):
+    SocialApp = apps.get_model('socialaccount', 'SocialApp')
+    Site = apps.get_model('sites', 'Site')
+    db_alias = schema_editor.connection.alias
+    app, created = SocialApp.objects.using(db_alias).get_or_create(
+        provider='google',
+        defaults={
+            'name': 'Google',
+            'client_id': '559124132137-h9jjj9jm7762uq7u2mn8h8dki7re28tv.apps.googleusercontent.com',
+            'secret': '1iNc8DKhNXoey8jLs4y-E6MN',
```
Comment (peterbe) [inline atmo/users/migrations/0002_google_auth.py]: ???

**Comment by maurodoglio** [comment]
:shipit:

File: atmo/users/migrations/0002_google_auth.py (hunk 4 above)
Comment (jezdez) [inline atmo/users/migrations/0002_google_auth.py]: I've replaced it with a make task and management command, thanks for the heads-up

**Pull request merged**